src/olympia/editors/models.py

```diff
@@ -704,10 +706,46 @@ class AutoApprovalSummary(ModelBase):
     verdict = models.PositiveSmallIntegerField(
         choices=amo.AUTO_APPROVAL_VERDICT_CHOICES,
         default=amo.NOT_AUTO_APPROVED)
+    weight = models.IntegerField(default=0)
 
     def __unicode__(self):
         return u'%s %s' % (self.version.addon.name, self.version)
 
+    def calculate_weight(self):
+        """Calculate the weight value for this version according to various
+        risk factors.
+
+        That value is then used in editor tools to prioritize add-ons in the
+        auto-approved queue."""
+        # Note: for the moment, some factors are in direct contradiction with
+        # the rules determining whether or not an add-on can be auto-approved
+        # in the first place, but we'll relax those rules as we move towards
+        # post-review.
+        addon = self.version.addon
+        one_year_ago = (self.created or datetime.now()) - timedelta(days=365)
```

Comment:
The `or` is because that method is called before we save the `AutoApprovalSummary` instance and therefore don't necessarily have `created` set. I could have used `datetime.now()` only, but I found it useful to rely on `created` if we recalculate weights later (which we'll definitely do at some point - that's why I've added a task for it).

I realize now that only one of the two cases are tested - the current tests use an unsaved instance, I'll add one with a saved instance, do test that it works when `created` is set.